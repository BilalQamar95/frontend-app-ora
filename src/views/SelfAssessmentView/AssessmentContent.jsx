import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';

import Prompt from 'components/Prompt';
import TextResponse from 'components/TextResponse';
import FileUpload from 'components/FileUpload';

import messages from './messages';

const AssessmentContent = ({
  submission,
  oraConfigData,
}) => {
  const { formatMessage } = useIntl();

  return (
    <div>
      <div className="d-flex justify-content-between">
        <h2 className="mb-4">{formatMessage(messages.yourResponse)}</h2>
      </div>
      <p>
        <strong>{formatMessage(messages.instructions)}: </strong>
        {formatMessage(messages.instructionsText)}
      </p>
      {oraConfigData.prompts.map((prompt, index) => (
        // eslint-disable-next-line react/no-array-index-key
        <div key={index}>
          <Prompt prompt={prompt} />
          <TextResponse
            submissionConfig={oraConfigData.submissionConfig}
            value={submission.response.textResponses[index]}
            isReadOnly
          />
        </div>
      ))}
      <FileUpload
        uploadedFiles={submission.response?.uploadedFiles}
        isReadOnly
      />
    </div>
  );
};

AssessmentContent.propTypes = {
  submission: PropTypes.shape({
    response: PropTypes.shape({
      textResponses: PropTypes.arrayOf(PropTypes.string),
      uploadedFiles: PropTypes.arrayOf(
        PropTypes.shape({
          fileDescription: PropTypes.string,
          fileName: PropTypes.string,
          fileSize: PropTypes.number,
        }),
      ),
    }),
  }).isRequired,
  oraConfigData: PropTypes.shape({
    prompts: PropTypes.arrayOf(PropTypes.string),
    // eslint-disable-next-line react/forbid-prop-types
    submissionConfig: PropTypes.any,
  }).isRequired,
};

export default AssessmentContent;
