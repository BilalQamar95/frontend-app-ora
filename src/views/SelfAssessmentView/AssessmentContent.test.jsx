import { shallow } from '@edx/react-unit-test-utils';
import AssessmentContent from './AssessmentContent';

jest.mock('@edx/paragon/icons', () => ({
  CheckCircle: 'CheckCircle',
}));

jest.mock('components/Prompt', () => 'Prompt');
jest.mock('components/TextResponse', () => 'TextResponse');
jest.mock('components/FileUpload', () => 'FileUpload');

describe('<AssessmentContent />', () => {
  const props = {
    submission: {
      response: {
        textResponses: ['test'],
        uploadedFiles: [{
          fileDescription: 'test',
        }],
      },
    },
    oraConfigData: {
      prompts: ['<p>test</p>'],
      submissionConfig: {
        maxFileSize: 100,
      },
    },
  };

  describe('render', () => {
    test('default', () => {
      const wrapper = shallow(<AssessmentContent {...props} />);
      expect(wrapper.snapshot).toMatchSnapshot();

      expect(wrapper.instance.findByType('Prompt')).toHaveLength(1);
    });

    test('no prompts', () => {
      const wrapper = shallow(<AssessmentContent {...props} oraConfigData={{ prompts: [] }} />);
      expect(wrapper.snapshot).toMatchSnapshot();

      expect(wrapper.instance.findByType('Prompt')).toHaveLength(0);
    });
  });
});
