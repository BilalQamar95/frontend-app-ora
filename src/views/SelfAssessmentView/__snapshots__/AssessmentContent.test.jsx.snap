// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssessmentContent /> render default 1`] = `
<div>
  <div
    className="d-flex justify-content-between"
  >
    <h2
      className="mb-4"
    >
      Your response
    </h2>
  </div>
  <p>
    <strong>
      Instructions
      : 
    </strong>
    Create a response to the prompt below.
    Progress will be saved automatically and you can return to complete your
    progress at any time. After you submit your response, you cannot edit
    it.
  </p>
  <div
    key="0"
  >
    <Prompt
      prompt="<p>test</p>"
    />
    <TextResponse
      isReadOnly={true}
      submissionConfig={
        Object {
          "maxFileSize": 100,
        }
      }
      value="test"
    />
  </div>
  <FileUpload
    isReadOnly={true}
    uploadedFiles={
      Array [
        Object {
          "fileDescription": "test",
        },
      ]
    }
  />
</div>
`;

exports[`<AssessmentContent /> render no prompts 1`] = `
<div>
  <div
    className="d-flex justify-content-between"
  >
    <h2
      className="mb-4"
    >
      Your response
    </h2>
  </div>
  <p>
    <strong>
      Instructions
      : 
    </strong>
    Create a response to the prompt below.
    Progress will be saved automatically and you can return to complete your
    progress at any time. After you submit your response, you cannot edit
    it.
  </p>
  <FileUpload
    isReadOnly={true}
    uploadedFiles={
      Array [
        Object {
          "fileDescription": "test",
        },
      ]
    }
  />
</div>
`;
